classes/AddressChecksumCore.php

```
@@ -45,12 +46,25 @@ public function generateChecksum($address)
         }
 
         $uniqId = '';
-        $fields = $address->getFields();
+        try {
+            $fields = $address->getFields();
+        } catch (Exception $e) {
+            throw new Exception($this->getTranslator()->trans('Your address is incomplete, please update it.', array(), 'Shop.Notifications.Error'));
```

When re-throwing exceptions, the previous exception should should always be included in the new one (third parameter on the Exception constructor), so that the original exception can be traced back.

---

I don't think this is the right place to set up a translated message. You should throw a specific Exception type which can be catched at a higher level (controller or business class), which in turn will display the appropriate message.